[Exchange](legacy tokens) How to get list of add-ins using Exchange tokens (#5068)
OfficeDev/office-js-docs-pr: docs/outlook/turn-exchange-tokens-on-off.md

PS C:\>
```

> An update is being deployed that enables a report from the last seven days of which add-ins requested an Exchange token, and whether the request was allowed or blocked. If your tenant has this update, you'll see a list of requests displayed in the **Allowed** or **Blocked** list. If an add-in was granted the token request, it appears in the **Allowed** list along with the date of the request. If the token request was denied, it appears in the **Blocked** list. It's possible for the same add-in to appear in both lists. This happens if the add-in was allowed to get tokens, but then tokens were turned off. The following example shows an add-in was blocked when it requested a token on February 25th.

Reviewer: ```suggestion
We are deploying an update that enables a report of which add-ins requested an Exchange token in the last seven days, and whether the request was allowed or blocked. If your tenant has this update, you'll see a list of requests displayed in the **Allowed** or **Blocked** list. If an add-in was granted the token request, it appears in the **Allowed** list along with the date of the request. If the token request was denied, it appears in the **Blocked** list. It's possible for the same add-in to appear in both lists. This happens if the add-in was allowed to get tokens, but then tokens were turned off. The following example shows an add-in was blocked when it requested a token on February 25th.
```

```console
PS C:\> Get-AuthenticationPolicy -AllowLegacyExchangeTokens
AllowLegacyExchangeTokens: False
Allowed: []
Blocked:
[
        { "49d3b812-abda-45b9-b478-9bc464ce5b9c" : "2025-02-25" }
]
PS C:\>
```

To identify any add-ins that were allowed or blocked Exchange tokens, use the `Get-App` command as shown in the following example.

```console
PS C:\> Get-App -Identity 49d3b812-abda-45b9-b478-9bc464ce5b9c | Select-Object -Property DisplayName, AppVersion, MarketplaceAssetID, ProviderName

DisplayName            AppVersion MarketplaceAssetID ProviderName
-----------            ---------- ------------------ ------------
Script Lab for Outlook 4.0.0.0    WA200001603        Microsoft
```

> The previous Script Lab example used the **Get a user identity token** sample that uses the `getUserIdentityTokenAsync` function to make the request.

Reviewer: ```suggestion
The previous Script Lab example uses the **Get a user identity token** sample and the `getUserIdentityTokenAsync` function to make the request.
```

PS C:\>
```

> If any add-ins requested a legacy Exchange Online token in the last seven days, it will appear in the **Allowed** or **Blocked** list. If an add-in was granted the token request, it appears in the **Allowed** list. If the token request was denied, it appears in the **Blocked** list. It is possible for the same add-in to appear in both lists. This can happen if they were allowed to get tokens, but then tokens were turned off.

Reviewer: ```suggestion
If any add-ins requested a legacy Exchange Online token in the last seven days, it will appear in the **Allowed** or **Blocked** list. If an add-in was granted the token request, it appears in the **Allowed** list. If the token request was denied, it appears in the **Blocked** list. It's possible for the same add-in to appear in both lists. This can happen if they were allowed to get tokens, but then tokens were turned off.
```
